packages/angular-cli/models/webpack-build-common.ts

```diff
@@ -23,21 +24,60 @@ export function getWebpackCommonConfig(
   const appRoot = path.resolve(projectRoot, appConfig.root);
   const appMain = path.resolve(appRoot, appConfig.main);
   const nodeModules = path.resolve(projectRoot, 'node_modules');
-  const styles = appConfig.styles
-               ? appConfig.styles.map((style: string) => path.resolve(appRoot, style))
-               : [];
-  const scripts = appConfig.scripts
-                ? appConfig.scripts.map((script: string) => path.resolve(appRoot, script))
-                : [];
-  const extraPlugins: any[] = [];
-
-  let entry: { [key: string]: string[] } = {
+
+  let extraPlugins: any[] = [];
+  let extraRules: any[] = [];
+  let lazyChunks: string[] = [];
+
+  let entryPoints: { [key: string]: string[] } = {
     main: [appMain]
   };
 
-  // Only add styles/scripts if there's actually entries there
-  if (appConfig.styles.length > 0) { entry['styles'] = styles; }
-  if (appConfig.scripts.length > 0) { entry['scripts'] = scripts; }
+  // process global scripts
+  if (appConfig.scripts.length > 0) {
+    const globalScrips = extraEntryParser(appConfig.scripts, appRoot, 'scripts');
+
+    // add entry points and lazy chunks
+    globalScrips.forEach(script => {
+      if (script.lazy) { lazyChunks.push(script.entry); }
+      entryPoints[script.entry] = (entryPoints[script.entry] || []).concat(script.path);
+    });
+
+    // load global scripts using script-loader
+    extraRules.push({
+      include: globalScrips.map((script) => script.path), test: /\.js$/, loader: 'script-loader'
```

Comment:
We always had `script-loader` for global scripts though, I merely added it conditionally instead of always having it.